test/functional/ui/input_spec.lua

```diff
@@ -361,25 +361,16 @@ describe('input non-printable chars', function()
     ]])
 
     feed('\005')
-    screen:expect([[
-      foobar                                                      |
-      {1:~                                                           }|
-      {3:                                                            }|
-      "Xtest-overwrite"                                           |
-      {9:WARNING: The file has been changed since reading it!!!}      |
-      {6:Do you really want to write to it (y/n)?}u                   |
-      {6:Do you really want to write to it (y/n)?}                    |
-      {6:Do you really want to write to it (y/n)?}^                    |
-    ]])
+    screen:expect_unchanged()
 
     feed('n')
     screen:expect([[
       foobar                                                      |
+      {1:~                                                           }|
       {3:                                                            }|
       "Xtest-overwrite"                                           |
       {9:WARNING: The file has been changed since reading it!!!}      |
       {6:Do you really want to write to it (y/n)?}u                   |
-      {6:Do you really want to write to it (y/n)?}                    |
       {6:Do you really want to write to it (y/n)?}n                   |
```

Comment:
For my understanding: why was only one of these "prompts"(?) removed?

Reply:
Old [implementation](https://github.com/neovim/neovim/pull/31525/files#diff-5134a5ea161c11fa8571f2cc0f9677d108ebda3e5741438147979ae504b7ef4aL58) emitted the message and pressed key on each invalid keypress. Now we stay inside the active cmdline state if the `command_line_handle_key()` switch returns early. Could probably make it behave the same if necessary.

Reply:
We now also return after keys that did not change the cmdline, so that the callers can still properly handle control characters. This test now displays `^E` rater than nothing.